File: baju-kita-mobile/lib/model/produk.dart
class Produk {
  final int id;
  final String name;
  final int price;
  final String bahan;
  final int stok;
  final int kategoriId;
  final String? image;

  Produk({
    required this.id,
    required this.name,
    required this.price,
    required this.bahan,
    required this.stok,
    required this.kategoriId,
    this.image,
  });

  factory Produk.fromJson(Map<String, dynamic> map) {
    return Produk(
      id: map['id'],
      name: map['name'],
      price: map['price'],
      bahan: map['bahan'],
      stok: map['stok'],
      kategoriId: map['kategori_id'],
      image: map['image'],
    );
  }
}

File: baju-kita-api/app/Models/Produk.php
<?php

namespace App\Models;

use App\Casts\PublicFileClient;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Produk extends Model
{
    use HasFactory;

    protected $guarded = ['id'];

    protected $casts = [
        'price' => 'integer',
        'stok' => 'integer',
        'image' => PublicFileClient::class,
        'kategori_id' => 'integer',
    ];

    public function kategori()
    {
        return $this->belongsTo(Kategori::class);
    }

    public function detail_transaksis()
    {
        return $this->hasMany(DetailTransaksi::class);
    }

    public function scopeOrderByBuyed(Builder $query)
    {
        return $query->withCount(['detail_transaksis' => fn($query) => $query
            ->whereNot('transaksi_id', null),
        ])
        ->orderByDesc('detail_transaksis_count');
    }
}
